projects/observability/src/shared/components/table/data-cell/log-timestamp/log-timestamp-table-cell-renderer.component.ts

```diff
@@ -0,0 +1,69 @@
+import { ChangeDetectionStrategy, Component, Inject, OnInit } from '@angular/core';
+import { DateCoercer } from '@hypertrace/common';
+import {
+  CoreTableCellParserType,
+  TableCellAlignmentType,
+  TableCellParserBase,
+  TableCellRenderer,
+  TableCellRendererBase,
+  TableColumnConfig,
+  TABLE_CELL_DATA,
+  TABLE_COLUMN_CONFIG,
+  TABLE_COLUMN_INDEX,
+  TABLE_DATA_PARSER,
+  TABLE_ROW_DATA
+} from '@hypertrace/components';
+import { LogEvent } from '@hypertrace/distributed-tracing';
+import { ObservabilityTableCellType } from '../../observability-table-cell-type';
+
+@Component({
+  selector: 'ht-log-timestamp-table-cell-renderer',
+  styleUrls: ['./log-timestamp-table-cell-renderer.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="log-timestamp" [htTooltip]="tooltip" [ngClass]="{ 'first-column': this.isFirstColumn }">
+      {{ this.duration }}
+    </div>
+    <ng-template #tooltip>{{ this.readableDateTime }}</ng-template>
+  `
+})
+@TableCellRenderer({
+  type: ObservabilityTableCellType.LogTimestamp,
+  alignment: TableCellAlignmentType.Left,
+  parser: CoreTableCellParserType.NoOp
+})
+export class LogTimestampTableCellRendererComponent extends TableCellRendererBase<string> implements OnInit {
+  public readonly spanStartTime: number;
+  public readonly timestamp?: string;
+  public readonly duration?: string;
+  public readonly readableDateTime?: string;
+
+  private readonly dateCoercer = new DateCoercer();
+
+  public constructor(
+    @Inject(TABLE_COLUMN_CONFIG) columnConfig: TableColumnConfig,
+    @Inject(TABLE_COLUMN_INDEX) index: number,
+    @Inject(TABLE_DATA_PARSER)
+    parser: TableCellParserBase<string, string, unknown>,
+    @Inject(TABLE_CELL_DATA) cellData: any,
+    @Inject(TABLE_ROW_DATA) rowData: LogEvent
+  ) {
+    super(columnConfig, index, parser, cellData, rowData);
+    this.spanStartTime = rowData.spanStartTime as number;
+    this.timestamp = rowData.timestamp;
+    this.readableDateTime =
+      this.timestamp
```

Comment:
is there a cleaner way to do this? whats the goal here? Why do we have a time string as input here?

Reply:
+1 - please use native/our existing utils like date coercer and TImeDuration. We never should be working with the strings directly. This one, for example we could take the difference and use our duration class.

Reply:
So currently we're getting a timestamp string which is in nano seconds for example `2021-05-04T12:13:55.543456T`

So all this logic is done to show cell value and tooltip in an expected format.

Reply:
Nanos shouldn't be relevant at this level - that's why we use the ISO format, so the UI is agnostic to the unit. The timestamp should render the same as all our other timestamps, right? Using the date coercer, then the display date pipe. The new part is the relative offset between the two timestamps (log event time and span start time), which would require taking the difference between the two and using our duration pipe (I'm not 100% sure that duration pipe is up to date for these requirements, but conceptually at least, that's what we should be going for)

Reply:
I have updated the `new look` for this in description.
Please do have a look!!